src/averageAges.js

```diff
@@ -55,7 +92,27 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const child = people.filter(
+    (person) => people.some(
+      (mom) => mom.name === person.mother));
+  const childAges = child.map(
+    (person) => person.born - people.find(
+      (mom) => mom.name === person.mother).born);
+  const diff = childAges.reduce(
+    (person, mom) => person + mom
+  ) / child.length;
+
+  const sons = people.filter(
+    person => people.some(
+      mom => person.mother === mom.name) && person.sex === 'm');
+  const sonsAges = sons.map(
+    son => son.born - people.find(
+      mom => mom.name === son.mother).born);
+  const sonsAgesDiff = sonsAges.reduce(
+    (son, mom) => son + mom
+  ) / sons.length;
+
```

Comment:
```suggestion
      const child = people.filter((person) => (
        people.some((mom) => mom.name === person.mother)
      ));

      const childAges = child.map((person) => (
        person.born - (
          people.find((mom) => mom.name === person.mother).born
        )
      ));

      const diff = childAges.reduce(
        (person, mom) => person + mom
      ) / child.length;
    
      const sons = people.filter(person => (
        people.some(
          mom => person.mother === mom.name
        ) && person.sex === 'm'
      ));

      const sonsAges = sons.map(son => (
        son.born - people.find(
          mom => mom.name === son.mother
        ).born
      ));

      const sonsAgesDiff = sonsAges.reduce(
        (son, mom) => son + mom
      ) / sons.length;

```
+/- пример нормального форматирования)
я тут по разному отформатировал, но все варианты ок, главый прицип это используй скобочки и закрывай скобочку в той же колонке где ты ее открыл